### PR TITLE
Initial commit of Proxy

### PR DESCRIPTION
**Pull request by mdupont**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,532 @@
+{
```
Comment (bogdanromanx) [inline package-lock.json]: what's the purpose of having both package and package-lock?
why are there more entries in the package-lock than in package?

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# Tini is now available at /sbin/tini
+ENTRYPOINT ["/sbin/tini", "--"]
+
+WORKDIR /usr/src/app
+COPY index.js ./
+COPY node_modules ./node_modules/
+
+# Default values for environment variables
+ENV NODE_ENV="production"
+ENV ELASTICSEARCH_CLIENT_URL=""
```
Comment (bogdanromanx) [inline docker/Dockerfile]: `http://localhost:9200` ?

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+# Tini is now available at /sbin/tini
+ENTRYPOINT ["/sbin/tini", "--"]
+
+WORKDIR /usr/src/app
+COPY index.js ./
+COPY node_modules ./node_modules/
+
+# Default values for environment variables
+ENV NODE_ENV="production"
```
Comment (bogdanromanx) [inline docker/Dockerfile]: do we still need these?

File: index.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const proxy = require('express-http-proxy');
+const app = require('express')();
+
+app.use('/', proxy(process.env.ELASTICSEARCH_CLIENT_URL));
+
+app.listen(8000);
```
Comment (bogdanromanx) [inline index.js]: can this be picked up from `process.env.???` as well?

File: package.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "nexus-search-webapp",
+  "version": "0.0.1",
+  "description": "Search web application for the Nexus Data Platform",
+  "main": "index.js",
+  "scripts": {
+    "start": "node index.js",
+    "test": "true"
```
Comment (bogdanromanx) [inline package.json]: `true`? is that normal?

File: package-lock.json (hunk 1 above)
Comment (mdupont) [reply package-lock.json]: package-lock contains exact versions of all dependencies and sub(.*)-dependencies, so you can have reproducible builds. package.json only contains the top-level ones, the ones you need.

File: docker/Dockerfile (hunk 3 above)
Comment (mdupont) [reply docker/Dockerfile]: We do in that case

File: index.js (hunk 4 above)
Comment (mdupont) [reply index.js]: If we pass it as an environment variable from the Dockerfile, yes. It has to be the same port the docker image exposes.

File: package.json (hunk 5 above)
Comment (mdupont) [reply package.json]: "test" is mandatory in a package.json, for now there is no test, so I fall back on that. Needs to be replaced by real unit tests once there is real code in the proxy.

**Pull request closed without merging**